javascript/react_js/react_testing_part_two.md

```diff
@@ -89,6 +95,8 @@ We mock the `onChange` handler using one of Jest's functions, `jest.fn()`. For t
 
 But what if you want to set up your mocks in a `beforeEach` block rather than in every test? That's fine in some cases. Though, having all of the setup for a test in the same block as the test itself makes it easier to understand any particular test as it eliminates the need to check the whole file for context. This makes the reviewing of subsequent changes in a project down the road substantially easier. Additionally, it decreases the chance of having leakage create problems throughout the test suite. Unless your test file is getting really long and the test prep itself is dozens of lines in length, default to setting up in each test case; otherwise, you may use `beforeEach`.
 
+It looks like `userEvent.setup()` is invoked every single test. As an exercise, create your own [setup function](https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent) to minimize code repetition.
```

Comment:
I think the link text could be made more descriptive here as well.

In addition I think that this is an important thing to bring up, but that it should be done in a slightly different way.

I do think we should make learners aware that they shouldn't put userEvent setup into a beforeEach, but rather than suggesting people write a custom setup function as an exercise we should just say that doing so is the proper way of shortening userEvent setup, and that we recommend having the setup in each test in the vast majority of cases.

Reply:
I was actually thinking about making it a learning exercise to encourage learners to read the docs and pick up the recommended idiomatic approach, but you make a good point. 

By the way, thank you for taking the time to give feedback. It means a lot to me as a first-time contributor.

Reply:
Not a problem, thank you for taking the time to make a contribution!